### PR TITLE
Add an event for health check failures.

### PR DESCRIPTION
**Pull request by brendandburns**

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1077,6 +1077,12 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 					containersToKeep[containerID] = empty{}
 					continue
 				}
+				ref, ok := kl.getRef(containerID)
+				if !ok {
+					glog.Warningf("No ref for pod '%v' - '%v'", ID, name)
```
Comment (bgrant0607) [inline pkg/kubelet/kubelet.go]: This can't possibly even compile.

File: pkg/kubelet/kubelet.go [hunk 2]
```diff
@@ -1077,6 +1077,12 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
...
+				} else {
+					record.Eventf(ref, "unhealthy", "Health Check Failed %v - %v", ID, name)
+				}
 				glog.V(1).Infof("pod %q container %q is unhealthy. Container will be killed and re-created.", podFullName, container.Name, live)
```
Comment (bgrant0607) [inline pkg/kubelet/kubelet.go]: These logs shouldn't be V(1).

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (brendandburns) [inline pkg/kubelet/kubelet.go]: I blame copy and paste.  fixed.

File: pkg/kubelet/kubelet.go (hunk 2 above)
Comment (brendandburns) [inline pkg/kubelet/kubelet.go]: done (although that was pre-existing...)

**Comment by brendandburns** [comment]
comments addressed.  ptal.

File: pkg/kubelet/kubelet.go [hunk 3]
```diff
@@ -1077,6 +1077,12 @@ func (kl *Kubelet) syncPod(pod *api.BoundPod, dockerContainers dockertools.Docke
 					containersToKeep[containerID] = empty{}
 					continue
 				}
+				ref, ok := kl.getRef(containerID)
+				if !ok {
+					glog.Warningf("No ref for pod '%v' - '%v'", ID, name)
+				} else {
+					record.Eventf(ref, "unhealthy", "Health Check Failed %v - %v", ID, name)
```
Comment (bgrant0607) [inline pkg/kubelet/kubelet.go]: Please don't use the term "Health Check". We now have both liveness and readiness probes in the API. You're reporting that the liveness probe failed.

**Comment by bgrant0607** [comment]
Please see my now-hidden comment: https://github.com/GoogleCloudPlatform/kubernetes/pull/4408#discussion_r24645257

File: pkg/kubelet/kubelet.go (hunk 3 above)
Comment (brendandburns) [inline pkg/kubelet/kubelet.go]: done.

**Comment by brendandburns** [comment]
oops, comments crossed paths on the wire.  addressed.

**Comment by bgrant0607** [comment]
LGTM

**Comment by saad-ali** [comment]
@brendandburns Out of an abundance of caution I reverted this change because it caused v1.3 to fail on both Shippable and Travis. Rerunning, [Travis is](https://travis-ci.org/GoogleCloudPlatform/kubernetes/builds/50650463) green, but [Shippable v1.3] (https://app.shippable.com/builds/54de25cbc44f060e00737a50) is still failing. The error seems transient. Could you squash the commits and resubmit.

**Pull request merged**